src/diagnostics/core_diagnostics.jl

```diff
@@ -733,3 +733,372 @@ add_diagnostic_variable!(
+    error_diagnostic_variable("lwp", model)
+
+function compute_lwp!(
+    out,
+    state,
+    cache,
+    time,
+    moisture_model::T,
+) where {T <: Union{EquilMoistModel, NonEquilMoistModel}}
+    thermo_params = CAP.thermodynamics_params(cache.params)
+    if isnothing(out)
+        out = zeros(axes(Fields.level(state.f, half)))
+        Operators.column_integral_definite!(
+            out,
+            state.c.ρ .* (cache.precomputed.cloud_diagnostics_tuple.q_liq),
+        )
+        return out
+    else
+        Operators.column_integral_definite!(
+            out,
+            state.c.ρ .* (cache.precomputed.cloud_diagnostics_tuple.q_liq),
+        )
+    end
+end
+
+add_diagnostic_variable!(
+    short_name = "lwp",
+    long_name = "Liquid Water Path",
+    standard_name = "atmosphere_mass_content_of_cloud_liquid_water",
+    units = "kg m-2",
+    comments = """
+    The total mass of liquid water in cloud per unit area.
+    (not just the area of the cloudy portion of the column). It doesn't include precipitating hydrometeors.
+    """,
+    compute! = compute_lwp!,
+)
+
+###
+# Ice water path (2d)
+###
+compute_clivi!(out, state, cache, time) =
+    compute_clivi!(out, state, cache, time, cache.atmos.moisture_model)
+compute_clivi!(_, _, _, _, model::T) where {T} =
+    error_diagnostic_variable("clivi", model)
+
+function compute_clivi!(
+    out,
+    state,
+    cache,
+    time,
+    moisture_model::T,
+) where {T <: Union{EquilMoistModel, NonEquilMoistModel}}
+    thermo_params = CAP.thermodynamics_params(cache.params)
+    if isnothing(out)
+        out = zeros(axes(Fields.level(state.f, half)))
+        Operators.column_integral_definite!(
+            out,
+            state.c.ρ .* (cache.precomputed.cloud_diagnostics_tuple.q_ice),
+        )
+        return out
+    else
+        Operators.column_integral_definite!(
+            out,
+            state.c.ρ .* (cache.precomputed.cloud_diagnostics_tuple.q_ice),
+        )
+    end
+end
+
+add_diagnostic_variable!(
+    short_name = "clivi",
+    long_name = "Ice Water Path",
+    standard_name = "atmosphere_mass_content_of_cloud_ice",
+    units = "kg m-2",
+    comments = """
+    The total mass of ice in cloud per unit area.
+    (not just the area of the cloudy portion of the column). It doesn't include precipitating hydrometeors.
+    """,
+    compute! = compute_clivi!,
+)
+
+
+###
+# Vertical integrated dry static energy (2d)
+###
+function compute_dsevi!(out, state, cache, time)
+    thermo_params = CAP.thermodynamics_params(cache.params)
+    if isnothing(out)
+        out = zeros(axes(Fields.level(state.f, half)))
+        cp = CAP.cp_d(cache.params)
+        z = Fields.coordinate_field(state.c.ρ).z
+        g = CAP.grav(cache.params)
+        dse = cache.scratch.ᶜtemp_scalar
+        @. dse =
+            state.c.ρ * (
+                cp * TD.air_temperature.(thermo_params, cache.precomputed.ᶜts) +
+                cache.core.ᶜΦ
+            )
```

Comment:
This looks a little suspicious: because there are two broadcasts, one from `@.` and one from `TD.air_temperature.`

My gut feeling is that this should fail. Have you checked that this works? Is this part of a build that is run in CI?

Reply:
`@.` only adds dots where there aren't any already. The dot after `air_temperature` is redundant here, but it shouldn't fail.

Reply:
I'll remove `.` after function calls on `@.` lines to avoid redundancy. I just assumed `@.` was only broadcasting base operations but good to know it does more than that. I've been quickly testing that things run and checking timeseries plots if I changed code significantly and was fine.

Reply:
`@.` adds dots to nearly all function call expressions, and lots of other expressions too: https://github.com/JuliaLang/julia/blob/3d20a9210a59097b46ed2cbdcd1e87435873bcfa/base/broadcast.jl#L1236-L1239